simple_use_case/nextflow/simplecase.nf

```diff
@@ -15,52 +18,70 @@ process generateMesh {
     conda "../source/envs/preprocessing.yaml"
 
     input:
-    file geo from geo_ch
+    file geofile
 
     output:
-    file "unit_square.msh" into msh_ch
+    file "unit_square.msh" into mesh_msh
 
     """
-    gmsh -2 -order 1 -format msh2 $geo -o unit_square.msh
+    gmsh -2 -setnumber domain_size ${params.domainSize} $geofile -o unit_square.msh
     """
 }
 
 
 process convertToXDMF {
-    // convert any msh file to xdmf
+    // convert .msh file to .xdmf
 
     conda "../source/envs/preprocessing.yaml"
 
     input:
-    file msh from msh_ch
+    file mesh from mesh_msh
 
     output:
-    file "unit_square.h5" into h5_ch
-    file "unit_square.xdmf" into xdmf_ch
+    file "unit_square.h5" into mesh_h5
+    file "unit_square.xdmf" into mesh_xdmf
 
     """
-    meshio convert $msh unit_square.xdmf
+    meshio convert $mesh unit_square.xdmf
     """
 }
 
 
 process solvePoisson {
-    // solve poisson equation using fenics code
+    // solve poisson equation using fenics
 
     conda "../source/envs/processing.yaml"
 
     // multiple inputs from different channels
     input:
-    file fenics_code from fenics_ch
-    file meshdata from h5_ch
-    file mesh from xdmf_ch
+    file fenics_code
+    file meshdata from mesh_h5
+    file mesh from mesh_xdmf
 
     output:
-    file "poisson.pvd" into pvd_ch
-    file "poisson*.vtu" into vtu_ch
+    file "poisson.pvd" into poisson_pvd
+    file "poisson*.vtu" into poisson_vtu
+    stdout poisson_stdout
 
     """
-    python $fenics_code --mesh $mesh --degree ${params.degree} --outputfile poisson.pvd
+    python $fenics_code --mesh $mesh --degree 2 --outputfile poisson.pvd
+    """
+}
+
+
+process readNumberOfDofs {
+    input:
+    val x from poisson_stdout
+
+    output:
+    stdout number_of_dofs
+
+    """
+    #!/usr/bin/python3
+    s = '''${x.replaceAll("\\n", "&")}'''
```

Comment:
Out of curiousity: it doesn't work to simply have `s = str(${x})` and then do `s.split("\n")`?

Reply:
The problem was that any whitespace contained in the variable x was "expanded". I actually tried what you suggest, but the result was something like:
```python
s = str(Solving variational problem
Number of dofs used: 357
)
```
And this lead to indentation errors when executing the bash script written by nextflow.

Reply:
ok, makes sense. But I was more refering to the replaceAll step. If you put ${x}$ in triple quotes, then python doesn't care about newlines etc, no?

Reply:
Not so important anyway. Just makes it harder to understand when you read it, I guess most people that read it wonder why newline characters have to be replaced. If it is mandatory maybe we can also add a comment?

Reply:
> ok, makes sense. But I was more refering to the replaceAll step. If you put ${x}$ in triple quotes, then python doesn't care about newlines etc, no?

I tried that as well. To be honest, I don't understand why that did not work. 

> Not so important anyway. Just makes it harder to understand when you read it, I guess most people that read it wonder why newline characters have to be replaced. If it is mandatory maybe we can also add a comment?

I will spend some more minutes on understanding this and either change it or add a comment on why this is necessary.